plugin/types.go

```diff
@@ -58,7 +58,7 @@ type Parent interface {
 // Root represents the plugin root
 type Root interface {
 	Parent
-	Init() error
+	Init([]string) error
```

Comment:
I think this should be `Init(map[string]interface{}) error` to keep it generic.

Reply:
Definitely. This was a super-quick hack to test out fixing the issue with someone at the Contributor Summit. I'm halfway through reworking it to be more generic.

Reply:
Hmm, this may be tough with external plugins b/c their config isn't persisted after the `init` call.

Reply:
The idea would be that they pass what they care about along as part of `state`.

Reply:
Chicken-and-egg problem though: init is called before we know the name of the plugin. So these seem like they need to be two different calls.

Reply:
I guess it could be an environment variable we set after calling `init`. Just hard if you want to use that data during the `init` call.

Reply:
Think two different calls make more sense. Something like `<plugin_script> root` then `<plugin_script> init <root_path> <root_state> <config>` ? 

I wonder if we should just do `<plugin_script>` and require it to return the root? So if no args are passed, people know that means they need to return the root JSON. Then we maintain symmetry by telling people that `<plugin_script> <method>` will always be invoked on an entry.

Two separate calls are also nice because we can then say "we'll only invoke `init` if your root object implements it". Then people writing simple plugins don't have to worry about it.

Reply:
I'm really tempted to derive the name from the file name. It would make this quite a bit simpler.

Reply:
Yeah, I think that might be easier. I don't like having "name" being referenced twice (once from the file and once from the root JSON outputted by `init`), but it's less things we have to tell people and it does seem like naming the script as `<plugin_name>.<ext>` is idiomatic (e.g. the `sshfs.sh` example does this).